### PR TITLE
[snyk] Add snyk2nvd converter

### PR DESCRIPTION
**Pull request by n1chre**
This commit adds a snyk2nvd converter. The tool can download the feed
from snyk.io and convert it to NVD format.

Test: go build && go test

### CODE REVIEW

File: providers/snyk/converter/converter.go [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+
+func (items baseSnykItemConverter) Convert() []*dstschema.NVDCVEFeedJSON10DefCVEItem {
+	out := make([]*dstschema.NVDCVEFeedJSON10DefCVEItem, len(items))
+	for i, item := range items {
+		out[i] = item.Convert()
+	}
+	return out
+}
+
+type baseSnykItem struct {
```
Comment (skogtwin) [inline providers/snyk/converter/converter.go]: This should be in schema.

File: providers/snyk/jsonschema/jsonschema.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
+// Package jsonschema was auto-generated.
+// Command: jsonschema2go -gofmt -gopkg jsonschema -goptr snyk.json
+package jsonschema
```
Comment (skogtwin) [inline providers/snyk/jsonschema/jsonschema.go]: let's replace all specific structs with a generic one from above.

File: providers/snyk/jsonschema/jsonschema.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	Registry           string                  `json:"registry"`
+	Severity           string                  `json:"severity"`
+	Title              string                  `json:"title"`
+	URL                string                  `json:"url"`
+	VulnerableVersions []string                `json:"vulnerableVersions"`
+}
+
+// Snyk was auto-generated.
+// Source: snyk.js
+type Snyk struct {
```
Comment (skogtwin) [inline providers/snyk/jsonschema/jsonschema.go]: and make this `map[sting]baseSnykItem`

File: providers/snyk/converter/converter.go [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+		BaseMetricV3: &dstschema.NVDCVEFeedJSON10DefImpactBaseMetricV3{
+			CVSSV3: &dstschema.CVSSV30{
+				VectorString: item.CVSSV3,
+				BaseScore:    item.CvssScore,
+			},
+		},
+	}
+}
+
+func convertGo(src *srcschema.Snyk) baseSnykItemConverter {
```
Comment (skogtwin) [inline providers/snyk/converter/converter.go]: see comments below/above, there should be just one of these.

**Comment by n1chre** [comment]
Rewrote main per @skogtwin 's comments

**Comment by n1chre** [comment]
Resolved all comments @skogtwin 
What was changed:
- simplified schema: everything is in the same format, so dont use SnykGolang, SnykJava, ... just use SnykAdvisory; Snyk is now a map[language][]*Advisories
- converters: only need to convert this one, not do any shenanigans with converting first to basesnyk item and then convert that. I didn't touch the logic, just deleted dupllication

**Pull request merged**